test/integration/controller/jobset_controller_test.go

```diff
@@ -1589,6 +2030,32 @@ func jobActive(job *batchv1.Job) bool {
 	return active
 }
 
+// matchJobSetRestarts checks that the supplied jobset js has expectedCount
+// as the value of js.Status.Restarts.
+func matchJobSetRestarts(js *jobset.JobSet, expectedCount int32) {
+	gomega.Eventually(func() (int32, error) {
+		newJs := jobset.JobSet{}
+		if err := k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &newJs); err != nil {
+			return 0, err
+		}
+
+		return newJs.Status.Restarts, nil
+	}, timeout, interval).Should(gomega.BeComparableTo(expectedCount))
```

Comment:
Nit: I think it is better practice to use `gomega.Equal(...)` for primitive types, and only use `gomega.BeComparableTo(..)` for complex types like (lists of protos, etc) which need some kind of recursive evaluation.

Reply:
This is done.